docs/advanced/Channels.md

```diff
@@ -206,16 +211,26 @@ export function* watchOnPings() {
   const socketChannel = yield call(createSocketChannel, socket)
 
   while (true) {
-    const payload = yield take(socketChannel)
-    yield put({ type: INCOMING_PONG_PAYLOAD, payload })
-    yield fork(pong, socket)
+    try {
+      // An error from socketChannel will cause the saga jump to the catch block
+      const payload = yield take(socketChannel)
+      yield put({ type: INCOMING_PONG_PAYLOAD, payload })
+      yield fork(pong, socket)
+    } catch(err) {
```

Comment:
btw I didn't manage to find a test for such a case, should we add one?

Reply:
Sure, I'll add one.